### ISSUE TITLE
termination criterion for MappingIsoparametric.invF

### ISSUE THREAD

**Issue by gdmcbain** [kinnala/scikit-fem#249]
While investigating failures in `FacetBasis` for `MeshQuad` #244, which raised an error in `MappingIsoparametric.invF`, suspicion fell on the termination criterion for the Newton iteration:

https://github.com/kinnala/scikit-fem/blob/70aa668560e76925a35f87b1c659a8e8f8ebda23/skfem/mapping/mapping_isoparametric.py#L140

It was asked there:
> Should this use a norm rather than a sum?

Since the elements of the decrement can be signed, the sum can involve cancellation or even be negative, thereby falsely passing the test.

Perhaps something like
```python
if (np.linalg.norm(dX, 1, (0, 2)) < 1e-6).all():
```
captures the intention.

**Comment by gdmcbain** [comment]
For example, with
```python
m = MeshQuad()
e = ElementQuad1()
fb = FacetBasis(m, e, facets=[0])
x = fb.mapping.G(fb.X, find=fb.find)
Y = fb.mapping.invF(x, tind=fb.tind)
```
stepping through the last line with `pdb`, I found that `x` is
```
array([[[0.11270167, 0.5       , 0.88729833]],

       [[0.        , 0.        , 0.        ]]])
```
which are the quadrature points along the base of the unit square and at the first iteration, the Newton increment is `dX`
```python
array([[[-0.77459667,  0.        ,  0.77459667]],

       [[-1.        , -1.        , -1.        ]]])
```
of which the `np.sum` is `-3.0`, well less than the tolerance `1e-6` and so passes the termination test even though it's clearly a large decrement.

**Comment by gdmcbain** [comment]
However, if the above-proposed 1-norm criterion is used, a number of tests in the current suite fail!  I'll put this up as a WIP PR to demonstrate.

**Comment by gdmcbain** [comment]
I think that the failures here are to do with the acceptance criterion for `MappingIsoparametric.invF` #251, in which points exterior to the reference element are strictly excluded.  They are fixed by applying a tolerance #252.
